Task9

ghummel1 opened the pull request:
Basic function for Task9 (Average Degree) and documentation completed

Missing tests (also doesn't write to file)

@tzielin1

ghummel1 committed:
Documentation Added

Added documentation for avgDeg

tzielin1 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into TASK9

tzielin1 committed:
Modified avgDeg

added version with least parameters needed to output an avg degree of
list of MIs

tzielin1 committed:
Added avgDeg to RandomMonomialIdeals

ghummel1 committed:
Minor edtis to function

Changed some syntax to fix erors (like changing = to :=)

ghummel1 committed:
Minor edits to function

Commented out example in documentation because it causes an error,
changed ZZ to RR because output isn't always an integer

ghummel1 committed:
Fixed errors

Function now works and there are no errors with the example in
documentation

Sondzus commented:
@ghummel1  @tzielin1  some conflicts exist now; 
Also see post and follow-up  35@Piazza - ask @parkerjoncus  or @monicayun  to see their changes to avgDim and implement them here in avgDeg too. Thanks.

ghummel1 committed:
Merge branch 'master' into TASK9

ghummel1 committed:
Added Histogram

Added historgram, updated function

ghummel1 committed:
Made some changes to documentation

ghummel1 committed:
Updated and Fixed Documentation

ghummel1 committed:
Added a Test

Tested to make sure average was computing correctly

ghummel1 committed:
Added Tests

Added various tests, not working, need to be debugged

ghummel1 committed:
Tests

ghummel1 committed:
Trying to fix tests

ghummel1 committed:
Fixed tests

Tests are now working. commented out tests do not work

ghummel1 committed:
Fixing tests

ghummel1 committed:
Finished Tests

ghummel1 committed:
Merge branch 'master' into TASK9

ghummel1 committed:
Editing Documentation

ghummel1 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into TASK9

Commit:
changed names

changed names of randomGeneratingSet in documentation and examples

ghummel1 committed:
Finshed Documentation

Finshed the documentation for degStats and ShowDegreeTally

Commit:
one slash two t

ghummel1 committed:
Fixed errors in documentation

ghummel1 committed:
Merge branch 'master' into TASK9

Commit:
delete ,

Sondzus commented:
This is ready to be checked by someone - similar to dimStats but need to check method, doc, and tests. :)

ghummel1 committed:
Merge branch 'master' into TASK9

sonjapetrovic committed:
minor documentation changes

sonjapetrovic committed:
matching change in dimStats documentation (for output explanation)

sonjapetrovic committed:
no need to sum a sequence by hand; M2 can do it.

- made also a matching edit in dimStats (just like in degStats).
- deleted some old junk files sitting in this branch.

sonjapetrovic committed:
comment on tests.

Sondzus commented:
Package installs and checks properly. 
I made some small changes to documentation and a minor change in method. 
Left a comment in tests. 

Looking for others' input to see if we want to 'prettify' anything.

Sondzus commented:
@ghummel1  @monicayun @parkerjoncus @sdgpfo 
A thought comes to mind: by default, we only return the average dimension/degree in dimStats and degStats.
Should we also go ahead and return the standard deviation too? 
And then also ShowTally optionally (this part already done).

sdgpfo commented:
@Sondzus @ghummel1 @monicayun @parkerjoncus 
Thanks @Sondzus, I believe we discussed this in a prior meeting, and in my opinion, yes, we should return the standard deviation as well; returning a list or hashtable now, has the advantage that if we decide to return more info later we can easily add it without having to change any caller methods.

Sondzus commented:
Yes, thanks for that reminder @sdgpfo  - this makes sense. 

I'm just wondering: maybe we can go ahead and do stdev already? it shouldn't be too difficult, I believe. @ghummel1 @monicayun @parkerjoncus  up for trying it out? 

By the way, side comment to clarify: 
dimStats and degStats are returning - by default - only the average, but it is not returned merely as a number, for the exact same reason @sdgpfo  states in her comment:  the average is returned as an element in a 1-element sequence. Adding more outputs will then not affect any caller methods. 
(We chose not to go for a List or a Table but a Sequence instead, at least for now, as that's how I've seen M2 return two or more outputs before. If we change the type of output later it will be easy to replace-all so I'm not worried about that.)

Sondzus commented:
@ghummel1 @parkerjoncus Can we get your input on doing standard deviation for these two methods (see comment prior to this one)?

parkerjoncus commented:
@Sondzus so for standard deviation are we also going to make an optional input for if the user wants to see the standard deviation? Or are we going to return it anyway by default with the average.

Sondzus commented:
I'd vote for returning (mean, stdDev) by default. @parkerjoncus 

(And keeping Tally still optional.)

Commit:
Merge branch 'master' into TASK9

Commit:
added standard deviation to dimStats and degStats

Calculated and added the standard deviation to both degStats and
dimStats. Still need to add to documentation

Sondzus commented:
Great, thanks @parkerjoncus ! I see your code - looks like it's doing the correct thing.

Sondzus commented:
I left some comments on tests in a previous push; @ghummel1 : 
   -- SP would like to see some prettier tests - any ideas? e.g. why the first listOfIdeals, why not hand-code some nice ones in QQ[x,y,z]?

Sondzus committed:
Merge branch 'master' into TASK9

Commit:
standard dev in documentation

added standard deviation in the output description and in the general
description of the documentation for both dimstats and degstats

ghummel1 committed:
Changed 1 to 2 in Tests

ghummel1 committed:
Revert "Changed 1 to 2 in Tests"

This reverts commit 15385c191dd440fe5fab0d58b3fca72fcae19035.

Commit:
fixed examples

fixed example errors from standard deviation addition for both dimStats
and degStats. Also made assertions for the standard deviation on each
test.

ghummel1 committed:
Merge remote-tracking branch 'origin/TASK9' into TASK9

Commit:
added to example

added degStDev in the example so that there is no error

Commit:
added to dim example

added variables for the outputs in the example and then output the
variables and what is in them.

Commit:
final edits to tests

final edits to tests for dimStats and degStats so they are the same

Commit:
make it look nice
